airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java

```diff
@@ -59,6 +64,12 @@ public AirbyteConnectionStatus check(final JsonNode config) {
       final String outputSchema = namingResolver.getIdentifier(config.get(JdbcUtils.SCHEMA_KEY).asText());
       attemptSQLCreateAndDropTableOperations(outputSchema, database, namingResolver, sqlOperations);
       return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+    } catch (final ConnectionErrorException ex) {
```

Comment:
Can someone teach me about what's happening at this spot? We are throwing `ConnectionErrorException` from various spots. I assume that means "There is a config error" (which is different than the generic programming errors). So we catch it here.
The difference from this block and the generic one below is:
* A different message - that makes sense to me because we can do a better job instructing the user what's going on.
* `AirbyteTraceMessageUtility.emitConfigErrorTrace(ex, message);` - what does this do?

My overall hope is that this `ConnectionErrorException` ends up _not_ creating a sentry error and the below `Exception` one does. Is that the case?

Reply:
Yes, the code throws `ConnectionErrorException` in areas where a configuration error can arise

`AirbyteTraceMessage` is used to provide to the surface the stack trace using `AirbyteMessage` as a [delivery vehicle](https://docs.google.com/document/d/1ctrj3Yh_GjtQ93aND-WH3ocqGxsmxyC3jfiarrF6NY0/edit#heading=h.3wnatqs8mj02). To be quite frank, this was lesser so a decision made from my end an implementation that was carried from the previous revisions/reviews prior to my work

Agreed, that `ConnectionErrorException` should _ideally_ not be a sentry error, however how much value will be lost if this is excluded from Sentry? From what I was able to quickly search there's an ability to filter out certain [Errors](https://docs.sentry.io/product/data-management-settings/filtering/#error-message), can this be broadly applied where to filter the `ConnectionErrorException` into a folder so that it's not lost but rather just limited interaction and no github issues created?

Reply:
Is the `AirbyteTraceMessageUtility.emitConfigErrorTrace` that makes it go to Sentry? If so, do you know why that is not in the general `Exception` catching?  cc @pedroslopez

Reply:
@bleonard It looks like `emitConfigErrorTrace` is properly setting the FailureType to `config_error` - we're not currently doing anything with the tag, but it means we can use this tag to correctly exclude them from triggering alerts or otherwise. 

Sentry does have some server-side filtering to prevent the issue from showing up in sentry at all, but it's a feature of their Business plan (i believe we're on the Team plan)

Reply:
An alternative to sentry's server-side filtering would be in our implementation we can ignore anything that's a `config_error` failure type and don't send it to sentry

Reply:
I think regardless we should still wrap it in a trace message so that the failure reason is properly recorded and shown to users in the right place